plugins/babel/src/run-babel.ts

```diff
@@ -0,0 +1,44 @@
+import { ToolKitError } from '@dotcom-tool-kit/error'
+import { hookConsole } from '@dotcom-tool-kit/logger'
+import type { BabelOptions } from '@dotcom-tool-kit/types/src/schema/babel'
+import * as babel from '@babel/core'
+import fg from 'fast-glob'
+import { promises as fs } from 'fs'
+import path from 'path'
+import type { Logger } from 'winston'
+
+export async function runBabel(
+  logger: Logger,
+  options: BabelOptions,
+  transformOptions?: babel.TransformOptions
+): Promise<void> {
+  const fileGlob = options.files ?? 'src/**/*.js'
+  const files = await fg(fileGlob)
+  // Work out the root of the glob so we can strip this part of the path out of
+  // the outputted files.
+  // E.g., a glob of 'src/**/*.js'   = src/a/b.js -> lib/a/b.js
+  //       a glob of 'src/a/**/*.js' = src/a/b.js -> lib/b.js
+  const { base } = fg.generateTasks(fileGlob)[0]
+
+  const outputPath = options.outputPath ?? 'lib'
+  await fs.mkdir(outputPath, { recursive: true })
+
+  if (options.configFile) {
+    const { configFile } = options
```

Comment:
Question: will having this as an option allow users to pass in their own Babel options?

Reply:
We expect users to set Babel options via a config file. This setting just (optionally) tells Babel where to find that config file: one thing Babel that can't read from the config file itself!

Reply:
Just wondering because in the ticket it says:
-do not allow passing in arbitrary Babel options to the tasks! should use the default Babel config files only (this is handled by @babel/core already)

Reply:
This option just tells `@babel/core` where to find those default config files